### PR TITLE
Do not set self serve flag if lock assignment and user assigned

### PR DESCRIPTION
**Pull request by nolanpro**
Fixes https://processmaker.atlassian.net/browse/FOUR-3132

This was already added to 4.1.18

### CODE REVIEW

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://bugfix-four-3132-processmaker.ci.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://bugfix-four-3132-processmaker.ci.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://bugfix-four-3132-processmaker.ci.processmaker.net

File: ProcessMaker/Repositories/TokenRepository.php [hunk 1]
```diff
@@ -89,7 +89,26 @@ public function persistActivityActivated(ActivityInterface $activity, TokenInter
         $this->addUserToData($token->getInstance(), $user);
         $this->addRequestToData($token->getInstance());
         $token->user_id = $user ? $user->getKey() : null;
-        $token->is_self_service = $token->getAssignmentRule() === 'self_service' ? 1 : 0;
```
Comment (caleeli) [inline ProcessMaker/Repositories/TokenRepository.php:92]: I suggest that all this assignment logic should be in the `getNextUser()` method and here something like
```
    $token->is_self_service = $token->getAssignmentRule() === 'self_service' && $user === null ? 1 : 0;
```

**Review by caleeli — CHANGES_REQUESTED** [review]

File: ProcessMaker/Repositories/TokenRepository.php (hunk 1 above)
Comment (nolanpro) [reply ProcessMaker/Repositories/TokenRepository.php:92]: @caleeli You are right, the presence of $user combined with getAssignmentRule() is sufficient to determine the is_self_service flag. Updated.

**Review by caleeli — APPROVED** [review]

**Review by boliviacoca — APPROVED** [review]

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://bugfix-four-3132-processmaker.ci.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://bugfix-four-3132-processmaker.ci.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://develop-processmaker.ci.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://bugfix-four-3132-processmaker.ci.processmaker.net

**Pull request merged**